### PR TITLE
Quicksave and Quickload

### LINKED ISSUE #530: Add ability to save and restore robot locations to soccer
Currently, we can drag robots and reset the robot positions. It would be great if there was a way to save the current state of the robots and restore it in to allow easier testing plays and things in simulator.

### PR DESCRIPTION
**Pull request by chachmu**
Solves issue with Stop Robots not working. Adds quicksave and quickload functions under Simulation. Closes  #530

### CODE REVIEW

File: soccer/FieldView.hpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ class FieldView : public QWidget {
 
     void rotate(int value);
 
+    const Geometry2d::TransformMatrix& getTeamToWorld() { return _teamToWorld; }
```
Comment (justbuchanan) [inline soccer/FieldView.hpp]: Let's put a `const` in between `()` and `{` to indicate that calling the method doesn't modify `FieldView`.

File: soccer/MainWindow.cpp [hunk 2]
```diff
@@ -878,24 +877,81 @@ void MainWindow::on_actionStopRobots_triggered() {
...
-        r->mutable_vel()->set_x(0);
-        r->mutable_vel()->set_y(0);
-        r->set_w(0);
+        if (robot->visible) {
+            SimCommand::Robot* r = cmd.add_robots();
+            r->set_shell(robot->shell());
+            r->set_blue_team(processor()->blueTeam());
+            Geometry2d::Point newPos =
+                _ui.fieldView->getTeamToWorld() * robot->pos;
+            r->mutable_pos()->set_x(newPos.x);
```
Comment (justbuchanan) [inline soccer/MainWindow.cpp]: Any reason for setting position here?  If we just set the velocity to zero it should keep them wherever they are already.

File: soccer/MainWindow.hpp [hunk 3]
```diff
@@ -204,4 +208,6 @@ private Q_SLOTS:
     std::vector<QPushButton*> _logPlaybackButtons;
 
     RJ::Time _firstLogTimestamp = -1;
+
+    Packet::SimCommand quickLoadCmd;
```
Comment (justbuchanan) [inline soccer/MainWindow.hpp]: Rename this to `_quickLoadCmd` to match the naming conventions with the other instance variables.

File: soccer/MainWindow.cpp [hunk 4]
```diff
@@ -878,24 +877,81 @@ void MainWindow::on_actionStopRobots_triggered() {
...
+            r->set_w(0);
+        }
     }
     _ui.fieldView->sendSimCommand(cmd);
 }
 
+void MainWindow::on_actionQuicksaveRobotLocations_triggered() {
+    _ui.actionQuickloadRobotLocations->setEnabled(true);
+    quickLoadCmd.reset();
+    int i = 0;
```
Comment (justbuchanan) [inline soccer/MainWindow.cpp]: I don't think the `i` variable is used anywhere

**Comment by justbuchanan** [comment]
Nice work!  I made a few minor notes, but overall looks great!

File: soccer/MainWindow.cpp (hunk 2 above)
Comment (chachmu) [inline soccer/MainWindow.cpp]: For whatever reason SimCommand doesn't work if the position is not set. It just prints "Trying to override non-existent robot".

Comment (justbuchanan) [inline soccer/MainWindow.cpp]: Ok, sounds good.

**Pull request merged**